include/seqan3/io/sam_file/format_bam.hpp

```diff
@@ -333,24 +333,43 @@ inline void format_bam::read_alignment_record(stream_type & stream,
         if (!std::ranges::equal(stream_view | views::take_exactly_or_throw(4), std::string_view{"BAM\1"}))
             throw format_error{"File is not in BAM format."};
 
-        int32_t tmp32{};
-        read_field(stream_view, tmp32);
+        int32_t l_text{}; // length of header text including \0 character
+        int32_t n_ref{}; // number of reference sequences
+        int32_t l_name{}; // 1 + length of reference name including \0 character
+        int32_t l_ref{}; // length of reference sequence
 
-        if (tmp32 > 0) // header text is present
-            read_header(stream_view | views::take_exactly_or_throw(tmp32), header, ref_seqs);
+        read_field(stream_view, l_text);
+
+        if (l_text > 0) // header text is present
+            read_header(stream_view | views::take_exactly_or_throw(l_text), header, ref_seqs);
 
-        int32_t n_ref;
         read_field(stream_view, n_ref);
 
         for (int32_t ref_idx = 0; ref_idx < n_ref; ++ref_idx)
         {
-            read_field(stream_view, tmp32); // l_name (length of reference name including \0 character)
+            read_field(stream_view, l_name);
 
-            string_buffer.resize(tmp32 - 1);
-            std::ranges::copy_n(std::ranges::begin(stream_view), tmp32 - 1, string_buffer.data()); // copy without \0 character
+            string_buffer.resize(l_name - 1);
+            std::ranges::copy_n(std::ranges::begin(stream_view), l_name - 1, string_buffer.data()); // copy without \0 character
```

Comment:
In theory, this should copy the `\0` character, because l_name is 1 + length including \0.

Reply:
Not sure what you mean :)

Do want to copy that `\0` too?

Reply:
The comment says: We do not copy `\0`. 
The code says: We copy `\0`.

So follow-up would be to double check and if we do not copy the `\0` to add a comment as to why is that. Because `l_name` is one more than the length including `\0`, so `l_name - ` should include the `\0`.